bowbuffer.go

```diff
@@ -15,6 +16,168 @@ type Buffer struct {
+	case Boolean:
+		buf.Data = make([]bool, size)
+	case String:
+		buf.Data = make([]string, size)
+	default:
+		panic(fmt.Errorf("unsupported type '%s'", typ))
+	}
+	return buf
+}
+
+// Len returns the length of the Buffer
+func (b Buffer) Len() int {
+	switch data := b.Data.(type) {
+	case []int64:
+		return len(data)
+	case []float64:
+		return len(data)
+	case []bool:
+		return len(data)
+	case []string:
+		return len(data)
+	default:
+		panic(fmt.Errorf("unsupported type '%T'", b.Data))
+	}
+}
+
+// SetOrDrop sets the value `value` at index `i` by attempting a type conversion to the Buffer Type.
+// Set the bit in the Buffer nullBitmapBytes if the conversion succeeded, or clear it otherwise.
+func (b *Buffer) SetOrDrop(i int, value interface{}) {
+	var valid bool
+	switch v := b.Data.(type) {
+	case []int64:
+		v[i], valid = Int64.Convert(value).(int64)
+	case []float64:
+		v[i], valid = Float64.Convert(value).(float64)
+	case []bool:
+		v[i], valid = Boolean.Convert(value).(bool)
+	case []string:
+		v[i], valid = String.Convert(value).(string)
+	default:
+		panic(fmt.Errorf("unsupported type %T", v))
+	}
+
+	if valid {
+		bitutil.SetBit(b.nullBitmapBytes, i)
+	} else {
+		bitutil.ClearBit(b.nullBitmapBytes, i)
+	}
+}
+
+// SetOrDropStrict sets the value `value` at index `i` by attempting a type assertion to the Buffer Type.
+// Set the bit in the Buffer nullBitmapBytes if the type assertion succeeded, or clear it otherwise.
+func (b *Buffer) SetOrDropStrict(i int, value interface{}) {
+	var valid bool
+	switch v := b.Data.(type) {
+	case []int64:
+		v[i], valid = value.(int64)
+	case []float64:
+		v[i], valid = value.(float64)
+	case []bool:
+		v[i], valid = value.(bool)
+	case []string:
+		v[i], valid = value.(string)
+	default:
+		panic(fmt.Errorf("unsupported type %T", v))
+	}
+
+	if valid {
+		bitutil.SetBit(b.nullBitmapBytes, i)
+	} else {
+		bitutil.ClearBit(b.nullBitmapBytes, i)
+	}
+}
+
+// GetValue gets the value at index `i` from the Buffer
+func (b *Buffer) GetValue(i int) interface{} {
+	if bitutil.BitIsNotSet(b.nullBitmapBytes, i) {
+		return nil
+	}
+	switch v := b.Data.(type) {
+	case []int64:
+		return v[i]
+	case []float64:
+		return v[i]
+	case []bool:
+		return v[i]
+	case []string:
+		return v[i]
+	default:
+		panic(fmt.Errorf("unsupported type %T", v))
+	}
+}
+
+// Less returns whether the value at index `i` is less that the value at index `j`.
+func (b Buffer) Less(i, j int) bool {
+	switch v := b.Data.(type) {
+	case []int64:
+		return v[i] < v[j]
+	case []float64:
+		return v[i] < v[j]
+	case []string:
+		return v[i] < v[j]
+	case []bool:
+		return !v[i] && v[j]
+	default:
+		panic(fmt.Errorf("unsupported type %T", v))
+	}
+}
+
+// NewBufferFromCol returns a new Buffer created from the column at index `colIndex`.
+func (b *bow) NewBufferFromCol(colIndex int) Buffer {
+	data := b.Column(colIndex).Data()
+	switch b.ColumnType(colIndex) {
+	case Int64:
+		arr := array.NewInt64Data(data)
```

Comment:
We are doing almost the same operations in all code blocks. Can't we use polymorphism power to simplify code and avoid smells ? 

https://golangbot.com/polymorphism/ 
https://medium.com/extreme-programming/avoiding-type-assertions-in-go-6feaa8762c27

Reply:
We thought very hard on this already, and we couldn't find a better solution yet.
Maybe with the help of generics introduced in Go 1.18?
In this PR, the implementation doesn't change at all though, it was just moved from `bowbuffer.gen.go`.

Reply:
Maybe create an issue? :)

Reply:
Yes, I think you need to create an issue because the technical debt increases drastically each time we are using these patterns

Reply:
Issue created